scss/popup.scss

```diff
@@ -48,14 +48,52 @@ main {
   display: flex;
 }
 
-.profile-img {
-  flex: 0 0 auto;
-  align-self: flex-start;
-  margin-right: 10px;
-  height: 36px;
-  width: 36px;
-  border-radius: 18px;
-  object-fit: cover;
+.user-avatar {
```

Comment:
Let's pull this into its own file and import it. Below I propose a few tweaks that I think will improve the functionality and the code's legibility.

One issue I saw is that the dropdown will render if you hover far below the avatar. This is because the enclosing div takes up the space all the way to the bottom and we're watching for if any of that is being hovered over.

<img width="895" alt="image" src="https://user-images.githubusercontent.com/6589960/101082410-682c9800-3579-11eb-83b3-36c39e91db03.png">

If we wrap the `UserAvatar` in an extra div in the `user-avatar.tsx` file like so

```typescript
export function UserAvatar({ src, clickLogout }) {
  return (
    <div>
      <div className="user-avatar">
        <img className="profile-img" src={src} />
        <div className="dropdown-content">
          <div className="caret-up"></div>
          <button className="logout-btn" onClick={clickLogout}>
            Logout
          </button>
        </div>
      </div>
    </div>
  )
}
```

Then the `.user-avatar` div will just enclose just the area around the `.profile-img` and the `.dropdown-content` which is what we want.

Another thing I noticed is that the caret doesn't line up with the center of the profile image. We can use some scss variables and some simple math to handle this and clean up the code. Note the `margin-left: $profile-image-radius - $caret-radius;` in the code below.

Another trick is to use `opacity: 0` instead of `display: none` so that the element always has height/width even if it isn't displayed. This will make it so that hovering over it is possible even when it is not visible (opacity 0) to the user.

Finally, we can use nesting to reduce the number of times we need to mention the same class names making our code easier to read and refactor later if we need to. We also refer to the `.logout-btn` by its class name instead of its tag name which is a good practice to prevent styles from unintentionally targeting the wrong elements.

Hope this is useful!

```scss

$profile-image-radius: 18px;
$caret-radius: 12px;

.user-avatar {
  position: relative;
  z-index: 2;

  .profile-img {
    flex: 0 0 auto;
    align-self: flex-start;
    margin-right: 10px;
    height: 2 * $profile-image-radius;
    width: 2 * $profile-image-radius;
    border-radius: $profile-image-radius;
    object-fit: cover;
  }

  .caret-up {
    width: 0;
    height: 0;
    border-left: #{$caret-radius} solid transparent;
    border-right: #{$caret-radius} solid transparent;
    border-bottom: #{$caret-radius} solid var(--post-btn-background-color);
  }

  // Hide the dropdown content whenever not being hovered over
  &:not(:hover) .dropdown-content {
    opacity: 0;
  }

  .dropdown-content {
    position: absolute;
    margin-left: $profile-image-radius - $caret-radius;

    .logout-btn {
      display: inline-block;
      width: 92px;
      height: 46px;
      font-size: 1.2rem;
      padding: 12px 6px;
      font-weight: bold;
      color: var(--post-btn-text-color);
      background-color: var(--post-btn-background-color);
      border: none;

      &:hover {
        text-decoration: underline;
      }
    }
  }
}
```

Reply:
Great! thanks

Reply:
made change and resubmitted. thanks!